File: java/1-TwoSum.java
import java.util.HashMap;

class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] result = solution.twoSum(new int[]{2, 7, 4, 1}, 9);
        System.out.println(result[0] + " " + result[1]);
    }
}

class Solution {

    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (hash.containsKey(diff)) {
                return new int[] { hash.get(diff), i };
            }
            hash.put(nums[i], i);
        }
        return null;
    }
}

File: go/1-TwoSum.go
package main

import "fmt"

func main() {
	fmt.Println(twoSum([]int{2, 1, 5, 3}, 4)); // Expected output: [1, 3]
	fmt.Println(twoSum([]int{2, 7, 11, 15}, 9)); // Expected output: [0, 1]
	fmt.Println(twoSum([]int{7, 3, 9, 6, 1, 8, 2, 4}, 5)); // Expected output: [1, 6]
	fmt.Println(twoSum([]int{3, 5, 2, 3, 4, 3}, 6)); // Expected output: [0, 3]
}

func twoSum(nums []int, target int) []int {
	hash := make(map[int]int);
	for i := 0; i < len(nums); i++ {
		diff := target - nums[i];
		if _, ok := hash[diff]; ok {
			return []int{hash[diff], i}
		}
		hash[nums[i]] = i;
	}
	return nil;
}